internal/service/appstream/stack_user_association_test.go

```diff
@@ -0,0 +1,197 @@
+package appstream_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appstream"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tfappstream "github.com/hashicorp/terraform-provider-aws/internal/service/appstream"
+)
+
+func TestAccAppStreamStackUserAssociation_basic(t *testing.T) {
+	resourceName := "aws_appstream_stack_user_association.test"
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	authType := "USERPOOL"
+	rEmail := acctest.RandomEmailAddress("hashicorp.com")
```

Comment:
This should use a random domain name generated by `acctest.RandomDomainName()`, to avoid being delivered to an actual email address.
```suggestion
	domain := acctest.RandomDomainName()
	rEmail := acctest.RandomEmailAddress(domain)
```